### PR TITLE
Add remoteWrite custom HTTP headers

### PR DESCRIPTION
**Pull request by KrotanHill**
Adds the possibility to configure the custom HTTP headers introduced in Prometheus v2.25.0
Implements #3850

I tested this on a cluster sending a tenantID to [cortex](https://github.com/cortexproject/cortex) and it worked.

I tried to work as written in the documents. Pleas tell me, if I made something wrong.

**Release Note Template (will be copied)**

```release-note:enhancement
Add custom HTTP headers in remoteWrite-config
```

### CODE REVIEW

File: pkg/prometheus/promcfg.go [hunk 1]
```diff
@@ -1401,6 +1401,10 @@ func (cg *configGenerator) generateRemoteWriteConfig(version semver.Version, p *
 			{Key: "remote_timeout", Value: spec.RemoteTimeout},
 		}
 
+		if len(spec.Headers) > 0 {
```
Comment (brancz) [inline pkg/prometheus/promcfg.go:1404]: We need to protect this if additionally with the version being >=2.25.0

**Review by brancz — COMMENTED** [review]
Just one small thing on compatibility, otherwise lgtm!

File: pkg/prometheus/promcfg.go (hunk 1 above)
Comment (KrotanHill) [reply pkg/prometheus/promcfg.go]: Makes sense. Code is updated.

**Comment by brancz** [comment]
Thank you!

**Pull request merged**